pkg/apis/metalkube/v1alpha1/baremetalhost_types.go

```diff
@@ -51,17 +51,13 @@ const (
 	// StateRegistering means we are telling the backend about the host
 	StateRegistering ProvisioningState = "registering"
 
+	// StateMatchProfile means we are comparing the discovered details
+	// against known hardware profiles
+	StateMatchProfile ProvisioningState = "match profile"
+
 	// StateReady means the host can be consumed
 	StateReady ProvisioningState = "ready"
 
-	// StatePreparingToProvision means we are updating the host to
-	// receive its image
-	StatePreparingToProvision ProvisioningState = "preparing to provision"
```

Comment:
I wonder if the fact that that validation is also a separate step is an implementation detail? I honestly would have expected a provisioning tool API to have at least one entry point where *all* of the info could be passed in to have it start doing the work. The fact that we're making a bunch of calls before Ironic does any work seems like something we don't need to show the user.

That said, if we want a PrepareForProvisioning() method in the provisioner API, I could see how to implement it.

Reply:
I don't think it really matters. It's fine to treat it as an implementation detail. Since we already had it there anyway, I thought there may be some value to the user in distinguishing between "there's no way we're doing anything with that nonsense you gave us" vs. "we tried but Ironic says no" - it could narrow down where to start debugging.
It would only really matter one way or the other if it significantly simplified the code. e.g. if we fail in validation we might be able to go straight back into attempting provisioning again once data changes, but if it fails during provisioning itself we might have deprovision first? If that were the case then having separate states would allow us to handle those transitions in the top-level Reconcile state machine, which would reduce complexity overall.